ramp-tps/src/tps.rs

```diff
@@ -0,0 +1,71 @@
+use crate::notifier::Notifier;
+use log::*;
+use solana_client::perf_utils::{sample_txs, SampleStats};
+use solana_client::thin_client::ThinClient;
+use solana_sdk::timing::duration_as_s;
+use std::{
+    net::SocketAddr,
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc, RwLock,
+    },
+    thread::{Builder, JoinHandle},
+};
+
+pub struct Sampler {
+    client: Arc<ThinClient>,
+    exit_signal: Arc<AtomicBool>,
+    maxes: Arc<RwLock<Vec<(String, SampleStats)>>>,
+    handle: Option<JoinHandle<()>>,
+}
+
+impl Sampler {
+    pub fn new(rpc_addr: &SocketAddr) -> Self {
+        let (_, dummy_socket) = solana_net_utils::bind_in_range((8000, 10_000)).unwrap();
+        let dummy_tpu_addr = *rpc_addr;
+        let client = Arc::new(ThinClient::new(*rpc_addr, dummy_tpu_addr, dummy_socket));
```

Comment:
Yuck, we need to create a `ThinClient` instead of an `RpcClient`?

Reply:
Yeah it's gross, we really only need `RpcClient` to sample transactions. I'll upstream a fix at some point: https://github.com/solana-labs/solana/issues/7517